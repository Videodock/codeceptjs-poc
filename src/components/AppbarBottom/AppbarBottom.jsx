import React from 'react';
import { makeStyles } from '@material-ui/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import * as PropTypes from 'prop-types';
import styles from './AppbarBottom.styles';

const useStyles = makeStyles(styles);

export default function AppbarBottom ({ children, position }) {
  const classes = useStyles();

  return (
    <AppBar classes={{ positionFixed: classes.appbarBottom }} position={position}>
      <Toolbar classes={{ dense: classes.toolbarBottom }} variant="dense" >
        {children}
      </Toolbar>
    </AppBar>
  );
}

AppbarBottom.propTypes = {
  position: PropTypes.oneOf(['absolute', 'fixed', 'relative', 'static', 'sticky']),
};

AppbarBottom.defaultProps = {
  position: 'fixed',
};
