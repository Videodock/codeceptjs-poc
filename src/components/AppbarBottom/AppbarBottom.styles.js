import { createStyles } from '@material-ui/styles';

export default () => createStyles({
  appbarBottom: {
    top      : 'auto',
    bottom   : 0,
    boxShadow: '0px -3px 6px 0px rgba(0,0,0,0.3)',
  },
  toolbarBottom: {
    height        : 56,
    justifyContent: 'flex-end',
  },
});
