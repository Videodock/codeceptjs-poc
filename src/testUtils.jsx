import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@material-ui/styles';
import { I18nextProvider } from 'react-i18next';
import { ApolloProvider } from '@apollo/client';
import { BrowserRouter as Router } from 'react-router-dom';
import 'url-search-params-polyfill';
import i18n from '../src/i18n';
import client from './graphql';
import theme from './theme';
import store from './redux';

const wrapper = ({ children }) => (
  <ApolloProvider client={client}>
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <I18nextProvider i18n={i18n}>
          <Router>
            {children}
          </Router>
        </I18nextProvider>
      </ThemeProvider>
    </Provider>
  </ApolloProvider>
);

const customRender = (ui, options) => render(ui, { wrapper, ...options });

export { customRender as render };
