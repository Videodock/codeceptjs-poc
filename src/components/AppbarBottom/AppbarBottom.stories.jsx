import React from 'react';
import { storiesOf } from '@storybook/react';
import AppbarBottom from './AppbarBottom';

storiesOf('AppbarBottom', module)
  .add('basic', () => <AppbarBottom />);
