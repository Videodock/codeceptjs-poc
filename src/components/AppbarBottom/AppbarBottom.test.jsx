import React from 'react';
import { render } from '../../testUtils';
import AppbarBottom from './AppbarBottom';

describe('<AppbarBottom />', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<AppbarBottom />);

    expect(container.firstChild).toMatchSnapshot();
  });

  // add your unit tests here
});
