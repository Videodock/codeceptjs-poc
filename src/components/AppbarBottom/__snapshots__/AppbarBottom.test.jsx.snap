// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppbarBottom /> renders and matches snapshot 1`] = `
<header
  class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed makeStyles-appbarBottom-1 MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
>
  <div
    class="MuiToolbar-root MuiToolbar-dense makeStyles-toolbarBottom-2 MuiToolbar-gutters"
  />
</header>
`;
